drivers/dac/dac_ad569x.c

```diff
@@ -0,0 +1,239 @@
+/*
+ * Copyright (c) 2024 Jan Kubiznak <jan.kubiznak@deveritec.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdint.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/dac.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/byteorder.h>
+
+LOG_MODULE_REGISTER(dac_ad569x, CONFIG_DAC_LOG_LEVEL);
+
+#define DAC_AD569X_MAX_CHANNEL 0
+
+#define AD569X_CTRL_GAIN(x)  (FIELD_PREP(BIT(11), x))
+#define AD569X_CTRL_REF(x)   (FIELD_PREP(BIT(12), x))
+#define AD569X_CTRL_PD(x)    (FIELD_PREP(BIT_MASK(2) << 13, x))
+#define AD569X_CTRL_RESET(x) (FIELD_PREP(BIT(15), x))
+
+enum ad569x_command {
+	AD569X_CMD_WRITE = 0x10,
+	AD569X_CMD_UPDATE = 0x20,
+	AD569X_CMD_WRITE_AND_UPDATE = 0x30,
+	AD569X_CMD_CONFIGURE = 0x40,
+};
+
+struct ad569x_config {
+	struct i2c_dt_spec bus;
+	uint8_t resolution;
+	uint8_t gain;
+	uint8_t voltage_reference;
+	uint8_t power_down_mode;
+};
+
+enum {
+	AD569X_CTRL_GAIN_1 = 0,
+	AD569X_CTRL_GAIN_2 = 1,
+	AD569X_CTRL_GAIN_DEFAULT = AD569X_CTRL_GAIN_1,
+	AD569X_CTRL_GAIN_MAX = AD569X_CTRL_GAIN_2,
+};
+
+enum {
+	AD569X_CTRL_REF_INTERNAL = 0,
+	AD569X_CTRL_REF_EXTERNAL = 1,
+	AD569X_CTRL_REF_DEFAULT = AD569X_CTRL_REF_INTERNAL,
+	AD569X_CTRL_REF_MAX = AD569X_CTRL_REF_EXTERNAL,
+};
+
+enum {
+	AD569X_CTRL_PD_NORMAL = 0,
+	AD569X_CTRL_PD_1K_OUT_IMPEDANCE = 1,
+	AD569X_CTRL_PD_100K_OUT_IMPEDANCE = 2,
+	AD569X_CTRL_PD_3_STATE_OUT_IMPEDANCE = 3,
+	AD569X_CTRL_PD_DEFAULT = AD569X_CTRL_PD_NORMAL,
+	AD569X_CTRL_PD_MAX = AD569X_CTRL_PD_3_STATE_OUT_IMPEDANCE,
+};
+
+enum {
+	AD569X_CTRL_NO_RESET = 0,
+	AD569X_CTRL_PERFORM_RESET = 1,
+	AD569X_CTRL_RESET_DEFAULT = AD569X_CTRL_NO_RESET
+};
```

Comment:
I'd suggest to move all those enums to the Devicetree binding. It will also take care of the default values and checking of the max values at build-time.

See [here](https://docs.zephyrproject.org/latest/build/dts/api/api.html#c.DT_ENUM_IDX) for the related documentation.

Reply:
I changed it for all but reset as I believe there is no point in having a DTS property for it.

Reply:
Yes, makes sense.